cz4013_group_project/message.py

```diff
@@ -1,51 +1,63 @@
 import socket
+from abc import ABC, abstractmethod
 
 
-class Message:
-    def __init__(self, message_type_identifier: int):
-        self.message_type_identifier: int = message_type_identifier
+class Message(ABC):
```

Comment:
should we create a new class e.g MessageFactory to deal with the creation of Message and its child objects?